src/util.cc

```diff
@@ -2536,6 +2538,21 @@ bool_ get_string(cptr prompt, char *buf, int len)
 	return (res);
 }
 
+/* Get a safe filename */
+bool_ get_filename(cptr prompt, char *buf, int len)
+{
+	bool_ res;
+
+	res = get_string(prompt, buf, len);
+
+	if (!portable_name(buf))
```

Comment:
This won't work, unfortunately. It's prevent the use of "~" which some(!) of the "create a file" functions actually do respect. (It's also a bit silly to restrict arbitrarily to ASCII, though more understandable, I suppose.)